providers/aws-ssm/aws.go

```diff
@@ -0,0 +1,212 @@
+package awsssm
+
+import (
+	"context"
+	"strconv"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/ssm"
+	"github.com/open-feature/go-sdk/openfeature"
+)
+
+// SSMMetadataKey is the key used in flag metadata to store SSM result metadata
+const SSMMetadataKey = "SSMMetadata"
+
+type AWS struct {
+	client     SSMClient
+	decryption bool
+}
+
+type SSMClient interface {
+	GetParameter(ctx context.Context, params *ssm.GetParameterInput, opts ...func(*ssm.Options)) (*ssm.GetParameterOutput, error)
+}
+
+func NewAWSService(cfg aws.Config) (*AWS, error) {
```

Comment:
```suggestion
func newAWSService(cfg aws.Config) *AWS {
```
This function always returns nil.